kratos/elements/embedded_nodal_variable_calculation_element_simplex.cpp

```diff
@@ -0,0 +1,211 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+//
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "elements/embedded_nodal_variable_calculation_element_simplex.h"
+
+
+namespace Kratos
+{
+
+template <class TVarType>
+void EmbeddedNodalVariableCalculationElementSimplex<TVarType>::CalculateLocalSystem(
+    MatrixType &rLeftHandSideMatrix,
+    VectorType &rRightHandSideVector,
+    ProcessInfo &rCurrentProcessInfo)
+{
+    this->CalculateLeftHandSide(rLeftHandSideMatrix, rCurrentProcessInfo);
+    this->CalculateRightHandSide(rRightHandSideVector, rCurrentProcessInfo);
+}
+
+template <>
+void EmbeddedNodalVariableCalculationElementSimplex<double>::CalculateLeftHandSide(
+    MatrixType &rLeftHandSideMatrix,
+    ProcessInfo &rCurrentProcessInfo)
+{
+    // Check size
+    if (rLeftHandSideMatrix.size1() != 2 || rLeftHandSideMatrix.size2() != 2) {
+        rLeftHandSideMatrix.resize(2, 2, false);
+    }
+
+    // Initialize Left Hand Side matrix
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(2,2);
```

Comment:
I don't think this is necessry, you overwrite the values later

Reply:
True. Removed in all the implementations.